kedro/pipeline/node.py

```diff
@@ -397,38 +399,55 @@ def _run_with_dict(self, inputs: Dict[str, Any], node_inputs: Dict[str, str]):
 
     def _outputs_to_dictionary(self, outputs):
         def _from_dict():
-            if set(self._outputs.keys()) != set(outputs.keys()):
+            (result,), iterator = (
+                ((outputs,), None)
+                if not inspect.isgenerator(outputs)
+                else spy(outputs, 1)
+            )
```

Comment:
This statement (and the identical one in `_from_list`) took me a while to parse, I think it may be a bit clearer without the ternary operator, which would avoid packing / unpacking `outputs` in the first case. I'd be interested to hear other's views on this.
```suggestion
            if not inspect.isgenerator(outputs):
                result, iterator = outputs, None
            else:
                (result,), iterator = spy(outputs, 1)
```

Reply:
Fair point, I think it can be rewritten to be clearer indeed.